### PR TITLE
HDRP - Better handling of supported platform build and better message

### PR DESCRIPTION
**Pull request by sebastienlagarde**
### Purpose of this PR

Fix for https://fogbugz.unity3d.com/f/cases/1267734/
and discussed [here](https://unity.slack.com/archives/C6Y79CZM0/p1600775100048300) and [here](https://unity.slack.com/archives/C6Y79CZM0/p1599662912243800)

---
### Testing status

Tested with unsupported graphic API (opengl) on windows in the editor and tried building a project.
Verified that the error message is correct.

---
### Comments to reviewers

It still needs to be tested on mac (and linux ?) but i don't have such machines
and also test the repro steps of the fogbugz

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Utility/HDUtils.cs [hunk 1]
```diff
@@ -1101,20 +1116,30 @@ internal static void DisplayUnsupportedMessage(string msg)
...
 
-            string msg = "Platform " + currentPlatform + " with device " + graphicAPI + " is not supported with High Definition Render Pipeline, no rendering will occur";
-            DisplayUnsupportedMessage(msg);
+            string msg = "Platform " + currentPlatform + " with graphic API " + graphicAPI + " is not supported with HDRP";
+
+            // Display more information to the users when it should have use Metal instead of OpenGL
+            if (graphicAPI.StartsWith("OpenGL"))
+            {
+                if (SystemInfo.operatingSystem.StartsWith("Mac"))
+                    msg += ", use the Metal graphic API instead";
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Utility/HDUtils.cs:1135]: graphic -> graphics 
For this and the Vulkan one below

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Utility/HDUtils.cs [hunk 2]
```diff
@@ -1101,20 +1116,30 @@ internal static void DisplayUnsupportedMessage(string msg)
...
+                else if (SystemInfo.operatingSystem.StartsWith("Windows"))
+                    msg += ", use the Vulkan graphic API instead";
+            }
+
+            msg += ".\nPlease change the platform/device to a compatible one or remove incompatible graphics APIs in Project Settings > Player > Other Settings.";
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Utility/HDUtils.cs:1140]: .\nChange the platform/device to a compatible one or remove incompatible graphics APIs. To do this, go to Project Settings > Player > Other Settings and modify the respective Graphics APIs for <platform> list.

**Review by JordanL8 — CHANGES_REQUESTED** [review]
Rewrote some bits

**Review by JordanL8 — APPROVED** [review]

**Review by adrien-de-tocqueville — APPROVED** [review]

**Comment by sebastienlagarde** [comment]
Merging for 10.2

**Review by JMargevics — COMMENTED** [review]
image
I've verified iOS error message, and message about vulcan on windows. All good ✔

 Will install Linux and check it there too soon.

**Pull request merged**